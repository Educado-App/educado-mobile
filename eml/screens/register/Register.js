import React, { useEffect } from 'react';
import { View, TouchableWithoutFeedback, Keyboard } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import RegisterForm from '../../components/login/RegisterForm';
import { SafeAreaView } from 'react-native-safe-area-context';
import LogoBackButton from '../../components/login/LogoBackButton';
import Text from '../../components/general/Text';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import AsyncStorage from '@react-native-async-storage/async-storage';

const LOGIN_TOKEN = '@loginToken';

export default function Register() {

	const navigation = useNavigation();

	const checkLoginToken = async () => {
		try {
			const fetchedToken = await AsyncStorage.getItem(LOGIN_TOKEN);
			if (fetchedToken !== null) {
				navigation.navigate('HomeStack');
			}
		} catch (error) {
			console.log('Failed to fetch the login token from storage');
		}
	};

	useEffect(() => {
		checkLoginToken();
	}, []);

	return (
		<SafeAreaView className="flex-1 justify-start bg-secondary">
			<KeyboardAwareScrollView
				className="flex-1"
				resetScrollToCoords={{ x: 0, y: 0 }}
				scrollEnabled={true}
			>
				<TouchableWithoutFeedback onPress={Keyboard.dismiss}>
					<View>
						<View className="mt-10">
							<LogoBackButton navigationPlace={'Login'} />
						</View>
						<View className="mx-6">
							<View className="mt-8">
								<RegisterForm />
							</View>
							<View className="flex-row justify-center items-end">
								<Text className="text-projectGray leading-5 text-base">
									{/* Already have an account? */}
                  JÃ¡ possui conta?
								</Text>
								<Text
									testId={'loginNav'}
									className={'text-projectBlack leading-5 text-base underline'}
									onPress={() => navigation.navigate('Login')}
								>
									{/* Log in now */}
                  Entre agora
								</Text>
							</View>
						</View>
					</View>
				</TouchableWithoutFeedback>
			</KeyboardAwareScrollView>
		</SafeAreaView>
	);
}
