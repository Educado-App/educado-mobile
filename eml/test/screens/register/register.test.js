import React from 'react';
import renderer from 'react-test-renderer';
import Register from '../../../screens/register/Register';
import AsyncStorage from '@react-native-async-storage/async-storage';

let navigated = false;
const mockToken = 'testToken';

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: jest.fn(() => { navigated = true; }),
  }),
}));

describe('Register screen', () => {

  let registerScreen;

  beforeEach(() => {
    navigated = false;
    AsyncStorage.clear();
    registerScreen = renderer.create(<Register />);
  });

  it('Login screen renders', () => {
    expect(registerScreen.toJSON()).toMatchSnapshot();
  });

  it('Pressing register new user navigates to the register page', async () => {
    const loginNav = registerScreen.root.findByProps({ testId: 'loginNav' });
    await renderer.act(() => {
      loginNav.props.onPress();
    });
    expect(navigated).toBe(true);
  });


  it('Check register when no valid token is stored', async () => {
    await renderer.act(() => {
      renderer.create(<Register />);
    });
    expect(navigated).toBe(false);
  });


  it('Check register when valid token stored', async () => {
    AsyncStorage.setItem('@loginToken', mockToken).then(async () => {
      await renderer.act(() => {
        renderer.create(<Register />);
      });
      expect(navigated).toBe(true);
    });
  });
});


/* TODO: Fix tests with AsyncStorage */ /*
test('Check login when valid token stored', async () => {
  const mockToken = "testToken";
  AsyncStorage.setItem("@loginToken", mockToken).then(async () => {
    await AsyncStorage.getItem("@loginToken");
    renderer.create(<Login />);
    expect(useNavigation().navigate).toHaveBeenCalledTimes(1);
  });
})*/