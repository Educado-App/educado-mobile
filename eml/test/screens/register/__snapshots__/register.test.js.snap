// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register screen Login screen renders 1`] = `
<RNCSafeAreaView
  edges={
    [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "justifyContent": "flex-start",
      },
      {
        "backgroundColor": "#F1F9FB",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        [
          {
            "marginTop": 40,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "width": "100%",
            },
            {
              "marginTop": 16,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "position": "absolute",
              },
              {
                "left": 0,
              },
              {
                "zIndex": 50,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 20,
                },
                undefined,
                false,
              ]
            }
            testID="RNE_BUTTON_WRAPPER"
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "invisible",
                    "borderColor": "#2089dc",
                    "borderRadius": 20,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "height": 50,
                    "justifyContent": "center",
                    "padding": 4,
                    "paddingHorizontal": 6,
                    "width": 50,
                  }
                }
              >
                <View
                  style={
                    {
                      "overflow": "hidden",
                    }
                  }
                  testID="RNE__ICON__CONTAINER"
                >
                  <View
                    accessibilityRole="button"
                    android_ripple={
                      {
                        "borderless": false,
                        "color": "rgba(0, 0, 0, 0.3)",
                        "radius": -5,
                      }
                    }
                    disabled={false}
                    testID="RNE__ICON__CONTAINER_ACTION"
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                      testID="RNE__ICON"
                    >
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "alignItems": "center",
              },
              {
                "justifyContent": "center",
              },
              {
                "width": "100%",
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              [
                {
                  "height": 48,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "marginLeft": 24,
            "marginRight": 24,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "marginTop": 32,
            },
          ]
        }
      >
        <View>
          <View
            style={
              [
                {
                  "marginBottom": 24,
                },
              ]
            }
          >
            <View>
              <View
                style={
                  [
                    {
                      "display": "flex",
                    },
                    {
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Montserrat-Regular",
                      },
                      {
                        "color": "#383838",
                      },
                      {
                        "fontSize": 16,
                      },
                      [
                        {
                          "marginLeft": 8,
                        },
                        {
                          "fontSize": 12,
                          "lineHeight": 16,
                        },
                      ],
                    ]
                  }
                >
                  Nome
                </Text>
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Montserrat-Regular",
                      },
                      {
                        "color": "#383838",
                      },
                      {
                        "fontSize": 16,
                      },
                      [
                        {
                          "marginLeft": 4,
                        },
                        {
                          "fontSize": 12,
                          "lineHeight": 16,
                        },
                        {
                          "color": "#FF4949",
                        },
                      ],
                    ]
                  }
                >
                  *
                </Text>
              </View>
              <View>
                <TextInput
                  autoComplete="off"
                  keyboardType="default"
                  onChangeText={[Function]}
                  passwordGuidelines={false}
                  placeholder="Nome"
                  secureTextEntry={false}
                  style={
                    [
                      {
                        "paddingBottom": 4,
                        "paddingTop": 4,
                      },
                      {
                        "paddingLeft": 10,
                      },
                      {
                        "backgroundColor": "#FFFFFF",
                      },
                      {
                        "borderBottomLeftRadius": 8,
                        "borderBottomRightRadius": 8,
                        "borderTopLeftRadius": 8,
                        "borderTopRightRadius": 8,
                      },
                    ]
                  }
                  value=""
                />
              </View>
            </View>
          </View>
          <View
            style={
              [
                {
                  "marginBottom": 24,
                },
              ]
            }
          >
            <View>
              <View
                style={
                  [
                    {
                      "display": "flex",
                    },
                    {
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Montserrat-Regular",
                      },
                      {
                        "color": "#383838",
                      },
                      {
                        "fontSize": 16,
                      },
                      [
                        {
                          "marginLeft": 8,
                        },
                        {
                          "fontSize": 12,
                          "lineHeight": 16,
                        },
                      ],
                    ]
                  }
                >
                  Sobrenome
                </Text>
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Montserrat-Regular",
                      },
                      {
                        "color": "#383838",
                      },
                      {
                        "fontSize": 16,
                      },
                      [
                        {
                          "marginLeft": 4,
                        },
                        {
                          "fontSize": 12,
                          "lineHeight": 16,
                        },
                        {
                          "color": "#FF4949",
                        },
                      ],
                    ]
                  }
                >
                  *
                </Text>
              </View>
              <View>
                <TextInput
                  autoComplete="off"
                  keyboardType="default"
                  onChangeText={[Function]}
                  passwordGuidelines={false}
                  placeholder="Sobrenome"
                  secureTextEntry={false}
                  style={
                    [
                      {
                        "paddingBottom": 4,
                        "paddingTop": 4,
                      },
                      {
                        "paddingLeft": 10,
                      },
                      {
                        "backgroundColor": "#FFFFFF",
                      },
                      {
                        "borderBottomLeftRadius": 8,
                        "borderBottomRightRadius": 8,
                        "borderTopLeftRadius": 8,
                        "borderTopRightRadius": 8,
                      },
                    ]
                  }
                  value=""
                />
              </View>
            </View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "center",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    {
                      "color": "#383838",
                    },
                    {
                      "fontSize": 16,
                    },
                    [
                      {
                        "fontSize": 12,
                        "lineHeight": 16,
                      },
                      {
                        "color": "#FF4949",
                      },
                      {
                        "marginLeft": 8,
                        "marginRight": 8,
                      },
                    ],
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              [
                {
                  "marginBottom": 24,
                },
              ]
            }
          >
            <View>
              <View
                style={
                  [
                    {
                      "display": "flex",
                    },
                    {
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Montserrat-Regular",
                      },
                      {
                        "color": "#383838",
                      },
                      {
                        "fontSize": 16,
                      },
                      [
                        {
                          "marginLeft": 8,
                        },
                        {
                          "fontSize": 12,
                          "lineHeight": 16,
                        },
                      ],
                    ]
                  }
                >
                  E-mail
                </Text>
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Montserrat-Regular",
                      },
                      {
                        "color": "#383838",
                      },
                      {
                        "fontSize": 16,
                      },
                      [
                        {
                          "marginLeft": 4,
                        },
                        {
                          "fontSize": 12,
                          "lineHeight": 16,
                        },
                        {
                          "color": "#FF4949",
                        },
                      ],
                    ]
                  }
                >
                  *
                </Text>
              </View>
              <View>
                <TextInput
                  autoComplete="off"
                  keyboardType="email-address"
                  onChangeText={[Function]}
                  passwordGuidelines={false}
                  placeholder="Insira sua e-mail"
                  secureTextEntry={false}
                  style={
                    [
                      {
                        "paddingBottom": 4,
                        "paddingTop": 4,
                      },
                      {
                        "paddingLeft": 10,
                      },
                      {
                        "backgroundColor": "#FFFFFF",
                      },
                      {
                        "borderBottomLeftRadius": 8,
                        "borderBottomRightRadius": 8,
                        "borderTopLeftRadius": 8,
                        "borderTopRightRadius": 8,
                      },
                    ]
                  }
                  value=""
                />
              </View>
            </View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "center",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    {
                      "color": "#383838",
                    },
                    {
                      "fontSize": 16,
                    },
                    [
                      {
                        "fontSize": 12,
                        "lineHeight": 16,
                      },
                      {
                        "color": "#FF4949",
                      },
                      {
                        "marginLeft": 8,
                        "marginRight": 8,
                      },
                    ],
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              [
                {
                  "marginBottom": 24,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "position": "relative",
                  },
                ]
              }
            >
              <View>
                <View
                  style={
                    [
                      {
                        "display": "flex",
                      },
                      {
                        "flexDirection": "row",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "fontFamily": "Montserrat-Regular",
                        },
                        {
                          "color": "#383838",
                        },
                        {
                          "fontSize": 16,
                        },
                        [
                          {
                            "marginLeft": 8,
                          },
                          {
                            "fontSize": 12,
                            "lineHeight": 16,
                          },
                        ],
                      ]
                    }
                  >
                    Senha
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "fontFamily": "Montserrat-Regular",
                        },
                        {
                          "color": "#383838",
                        },
                        {
                          "fontSize": 16,
                        },
                        [
                          {
                            "marginLeft": 4,
                          },
                          {
                            "fontSize": 12,
                            "lineHeight": 16,
                          },
                          {
                            "color": "#FF4949",
                          },
                        ],
                      ]
                    }
                  >
                    *
                  </Text>
                </View>
                <View>
                  <TextInput
                    autoComplete="off"
                    keyboardType="default"
                    onChangeText={[Function]}
                    passwordGuidelines={false}
                    placeholder="Insira sua senha"
                    secureTextEntry={true}
                    style={
                      [
                        {
                          "paddingBottom": 4,
                          "paddingTop": 4,
                        },
                        {
                          "paddingLeft": 10,
                        },
                        {
                          "backgroundColor": "#FFFFFF",
                        },
                        {
                          "borderBottomLeftRadius": 8,
                          "borderBottomRightRadius": 8,
                          "borderTopLeftRadius": 8,
                          "borderTopRightRadius": 8,
                        },
                      ]
                    }
                    value=""
                  />
                </View>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "position": "absolute",
                    },
                    {
                      "right": 0,
                    },
                    {
                      "top": 8,
                    },
                    {
                      "paddingBottom": 12,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 12,
                    },
                  ]
                }
              >
                <Text />
              </View>
            </View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "justifyContent": "flex-start",
                  },
                  {
                    "marginTop": 4,
                  },
                  {
                    "height": 24,
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    {
                      "color": "#383838",
                    },
                    {
                      "fontSize": 16,
                    },
                    [
                      {
                        "fontSize": 12,
                        "lineHeight": 16,
                      },
                      {
                        "color": "#A1ACB2",
                      },
                    ],
                  ]
                }
                testId="passwordLengthAlert"
              >
                • Mínimo 8 caracteres
              </Text>
              <View
                style={
                  [
                    {
                      "flexDirection": "row",
                    },
                    {
                      "alignItems": "center",
                    },
                    {
                      "transform": [
                        {
                          "translateY": -4,
                        },
                        {
                          "translateX": 0,
                        },
                      ],
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "justifyContent": "flex-start",
                  },
                  {
                    "height": 24,
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    {
                      "color": "#383838",
                    },
                    {
                      "fontSize": 16,
                    },
                    [
                      {
                        "fontSize": 12,
                        "lineHeight": 16,
                      },
                      {
                        "fontFamily": "Montserrat-Regular",
                      },
                      {
                        "color": "#A1ACB2",
                      },
                    ],
                  ]
                }
                testId="passwordLetterAlert"
              >
                • Conter pelo menos uma letra
              </Text>
              <View
                style={
                  [
                    {
                      "flexDirection": "row",
                    },
                    {
                      "alignItems": "center",
                    },
                    {
                      "transform": [
                        {
                          "translateY": -4,
                        },
                        {
                          "translateX": 0,
                        },
                      ],
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              [
                {
                  "marginBottom": 8,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "position": "relative",
                  },
                ]
              }
            >
              <View>
                <View
                  style={
                    [
                      {
                        "display": "flex",
                      },
                      {
                        "flexDirection": "row",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "fontFamily": "Montserrat-Regular",
                        },
                        {
                          "color": "#383838",
                        },
                        {
                          "fontSize": 16,
                        },
                        [
                          {
                            "marginLeft": 8,
                          },
                          {
                            "fontSize": 12,
                            "lineHeight": 16,
                          },
                        ],
                      ]
                    }
                  >
                    Confirmar senha
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "fontFamily": "Montserrat-Regular",
                        },
                        {
                          "color": "#383838",
                        },
                        {
                          "fontSize": 16,
                        },
                        [
                          {
                            "marginLeft": 4,
                          },
                          {
                            "fontSize": 12,
                            "lineHeight": 16,
                          },
                          {
                            "color": "#FF4949",
                          },
                        ],
                      ]
                    }
                  >
                    *
                  </Text>
                </View>
                <View>
                  <TextInput
                    autoComplete="off"
                    keyboardType="default"
                    onChangeText={[Function]}
                    passwordGuidelines={false}
                    placeholder="Confirme sua senha"
                    secureTextEntry={true}
                    style={
                      [
                        {
                          "paddingBottom": 4,
                          "paddingTop": 4,
                        },
                        {
                          "paddingLeft": 10,
                        },
                        {
                          "backgroundColor": "#FFFFFF",
                        },
                        {
                          "borderBottomLeftRadius": 8,
                          "borderBottomRightRadius": 8,
                          "borderTopLeftRadius": 8,
                          "borderTopRightRadius": 8,
                        },
                      ]
                    }
                    value=""
                  />
                </View>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "position": "absolute",
                    },
                    {
                      "right": 0,
                    },
                    {
                      "top": 8,
                    },
                    {
                      "paddingBottom": 12,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 12,
                    },
                  ]
                }
              >
                <Text />
              </View>
            </View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "center",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    {
                      "color": "#383838",
                    },
                    {
                      "fontSize": 16,
                    },
                    [
                      {
                        "fontSize": 12,
                        "lineHeight": 16,
                      },
                      {
                        "color": "#FF4949",
                      },
                      {
                        "marginLeft": 8,
                        "marginRight": 8,
                      },
                    ],
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              [
                {
                  "marginBottom": 8,
                  "marginTop": 8,
                },
              ]
            }
          >
            <View>
              <View
                accessibilityState={
                  {
                    "disabled": true,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "backgroundColor": "#5ECCE9",
                    "borderBottomLeftRadius": 8,
                    "borderBottomRightRadius": 8,
                    "borderTopLeftRadius": 8,
                    "borderTopRightRadius": 8,
                    "opacity": 0.5,
                    "paddingBottom": 16,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 16,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Montserrat-Regular",
                      },
                      {
                        "color": "#383838",
                      },
                      {
                        "fontSize": 16,
                      },
                      [
                        {
                          "textAlign": "center",
                        },
                        {
                          "fontFamily": "Montserrat-Bold",
                        },
                        {
                          "fontSize": 16,
                        },
                        {
                          "color": "#FFFFFF",
                        },
                      ],
                    ]
                  }
                >
                  Cadastrar
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "flex-end",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "color": "#A1ACB2",
                },
                {
                  "lineHeight": 20,
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
              ],
            ]
          }
        >
          Já possui conta?
        </Text>
        <Text
          onPress={[Function]}
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "color": "#383838",
                },
                {
                  "lineHeight": 20,
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                {
                  "textDecorationLine": "underline",
                },
              ],
            ]
          }
          testId="loginNav"
        >
          Entre agora
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
